### ISSUE TITLE
[Hi-lvl Python SDK] Add a convenient API for accessing revisions on a branch

### ISSUE THREAD

**Issue by nicholasjng** [treeverse/lakeFS#7232]
I find myself constantly needing to use something equivalent to `git rev-parse $BRANCH~n`, for n > 1. Getting that commit is already possible, but I have not found a way to make it nice.

Currently, I'm using something like `revision = list(Branch("hello").log(n))[-1]`, but that's not great for a number of reasons - it fetches all revisions up to `n`, and it requires me creating a list (or calling `next` n-1 times).

I'm aware that there are additional keyword arguments to `log()` which might be helpful here (amount and limit, probably), but I'd prefer an API that does not require unpacking the generator into a container first in user code.

In short:

```python

class Branch:

    def revision(n: int = 0) -> Commit:
        """Fetches the n-th ancestor of HEAD. Throws if HEAD has fewer than n ancestors."""
        ...
```

Do you think this would be possible to add? (I am aware of the merge-parent following issue, i.e. `$BRANCH~n` vs. `$BRANCH^n` - that might complicate things.)

**Comment by arielshaqed** [comment]
Hi @nicholasjng ,

I had a look and it seems that writing the code in terms of references works:
```py
>>> lakefs.Repository('repo').ref('main~2').get_commit()
Commit(id="40bc07fc089db2288763dacd297c2897209217fcc650f1039728df25220a137d")
```
It performs a single API call, too!

But if you couldn't figure it out, we need to do something about discoverability.

**Comment by nicholasjng** [comment]
Neat! This deserves an explanation in the docs, to clarify what can go in the ref constructor. Just for my information, you don't distinguish between references and revisions in lakeFS?

**Comment by arielshaqed** [comment]
> Neat! This deserves an explanation in the docs, to clarify what can go in the ref constructor.

I'll have a go at adding this to the PR.  I would really appreciate your comments on the PR.  The thing is that as developers we always assume all users know the [details of our architecture](https://docs.lakefs.io/understand/model.html#ref-expressions) at least as much as we do.[^1]

> Just for my information, you don't distinguish between references and revisions in lakeFS?

We mixed up "refs" as a shorthand for what the Git Book calls a [revision selector](https://git-scm.com/book/en/v2/Git-Tools-Revision-Selection).[^2]

| lakeFS | Git|
|---|---|
| ref or ref-expression | revision selector; most command docs say "&lt;commit>" instead |
| branch | branch |
| tag | tag |
| revision | revision |
| commit | commit (but _not_ "&lt;commit>") |

Sorry.

[^1]:  I can and do make excuses for this, just not good ones.
[^2]:  Of course I have excuses for this, "revision" was something else in a previous internal data model...
